linera-chain/src/certificate/generic.rs

```diff
@@ -20,6 +21,17 @@ pub struct GenericCertificate<T: CertificateValue> {
     signatures: Vec<(ValidatorPublicKey, ValidatorSignature)>,
 }
 
+impl<T: Allocative + CertificateValue> Allocative for GenericCertificate<T> {
+    fn visit<'a, 'b: 'a>(&self, visitor: &'a mut Visitor<'b>) {
+        visitor.visit_field(Key::new("GenericCertificate_value"), &self.value);
+        visitor.visit_field(Key::new("GenericCertificate_round"), &self.round);
+        for (public_key, _) in &self.signatures {
+            visitor.visit_field(Key::new("ValidatorPublicKey"), public_key);
```

Comment:
public keys also have a fixed size, don't they? Why assume the size for signatures but not public keys?

Reply:
You are right. Corrected.